subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2242,8 +2295,17 @@ static ssize_t send_tls(struct tls_context *ctx, const void *buf,
 				break;
 			}
 		} else {
-			(void)tls_mbedtls_reset(ctx);
-			errno = EIO;
+			/* MbedTLS API documentation requires session to
+			 * be reset in other error cases
+			 */
+			ret = tls_mbedtls_reset(ctx);
+			if (ret != 0) {
+				ctx->error = ENOMEM;
+				errno = ENOMEM;
+			}
+
+			ctx->error = ECONNABORTED;
+			errno = ECONNABORTED;
```

Comment:
Should the `ECONNABORTED` case be in the `else` case? Or maybe `ret` has to be set? In any case `ctx->error` is set twice.

Maybe add a helper function to reset the mbed TLS context?

Reply:
Yes, `else` was missing here, fixed. I don't think `ret` has to be set here, as we're not using it on return.